src/wallet/rpcwallet.cpp

```diff
@@ -2365,6 +2367,14 @@ UniValue listunspent(const UniValue& params, bool fHelp)
     if (params.size() > 1)
         nMaxDepth = params[1].get_int();
 
+    int nResultLimit = 0;
+    if (params.size() > 3) {
+        nResultLimit = params[3].get_int();
+        if (nResultLimit < 0) {
+            throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid parameter, limit must be >= 0.");
+        }
+    }
```

Comment:
Nit: I'd prefer this to be below `setAddresses`, so the parameters are parsed in positional order (following the style of all other RPC methods AFAIR).